### PR TITLE
Offline conferences

### PR DESCRIPTION
**Pull request by vargaat**
- Added conference data downloader.
- Updated conference details for offline mode.
- Fixed a crash when presenting offline alert from tap gesture recognizer.
- Fixed a warning about UI being modified from background thread.

refs: MBL-16769
affects: Student
release note: none

test plan:
- Sync the BigBlueButton tab for offline usage.
- Turn off internet on the device.
- Go to BigBlueButton page.
- List should show both ongoing and concluded conferences.
- Conference details page should have the join button disabled when visiting an active conference.

## Checklist

- [ ] Tested on phone
- [ ] Tested on tablet

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-16769](https://instructure.atlassian.net/browse/MBL-16769)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.39% | 92.42% | -0.03%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 25182df2ce7134a2c9947029817b6666c4b2d3b8
</p>

File: Core/Core/AppEnvironment/AppEnvironment.swift [hunk 1]
```diff
@@ -99,11 +99,25 @@ open class AppEnvironment {
...
-            controller = controller?.presentedViewController
+        let locateTopViewController: () -> UIViewController? = {
+            var controller = self.window?.rootViewController
+            while controller?.presentedViewController != nil {
+                controller = controller?.presentedViewController
+            }
+            return controller
+        }
+
+        if Thread.isMainThread {
```
Comment (idevainstr) [inline Core/Core/AppEnvironment/AppEnvironment.swift:110]: couldn't we just trade this whole block with:
`return DispatchQueue.main.sync {
    locateTopViewController()
}`

**Review by idevainstr — APPROVED** [review]
QA +1

File: Core/Core/AppEnvironment/AppEnvironment.swift (hunk 1 above)
Comment (vargaat) [reply Core/Core/AppEnvironment/AppEnvironment.swift:110]: Good idea. I didn't know there was such an overload.

**Review by szabinst — DISMISSED** [review]
QA + 1

**Review by szabinst — APPROVED** [review]

**Pull request merged**